caikit/core/model_management/local_model_trainer.py

```diff
@@ -0,0 +1,328 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The LocalModelTrainer uses a local thread to launch and manage each training job
+"""
+
+# Standard
+from concurrent.futures.thread import _threads_queues
+from datetime import datetime, timedelta
+from typing import Optional, Type
+import os
+import re
+import threading
+import uuid
+
+# First Party
+import aconfig
+import alog
+
+# Local
+from ..modules import ModuleBase
+from ..toolkit.destroyable_process import DestroyableProcess
+from ..toolkit.destroyable_thread import DestroyableThread
+from ..toolkit.errors import error_handler
+from .model_trainer_base import ModelTrainerBase
+import caikit
+
+log = alog.use_channel("LOC-TRNR")
+error = error_handler.get(log)
+
+
+OOM_EXIT_CODE = 137
+
+
+# 🌶️🌶️🌶️
+# Fix for python3.9, 3.10 and 3.11 issue where forked processes always exit with exitcode 1
+# when it's created inside a ThreadPoolExecutor: https://github.com/python/cpython/issues/88110
+# Fix taken from https://github.com/python/cpython/pull/101940
+# Credit: marmarek, https://github.com/marmarek
+
+if hasattr(os, "register_at_fork"):
+    os.register_at_fork(after_in_child=_threads_queues.clear)
+
+
+class LocalModelTrainer(ModelTrainerBase):
+    __doc__ = __doc__
+
+    name = "LOCAL"
+
+    class LocalModelFuture(ModelTrainerBase.ModelFutureBase):
+        """A local model future manages an execution thread for a single train
+        operation
+        """
+
+        def __init__(
+            self,
+            parent_name: str,
```

Comment:
unclear how many docstrings we need on this but `parent_name` is a bit opaque here.
Is this just the name of the trainer for the reversible hashing bit of the id? Would `trainer_name` be more understandable?

Reply:
Yeah, this is just the name of the trainer. I'll make that change (and in the base class)